jsonschema-core.xml

```diff
@@ -1687,14 +1692,64 @@
                             an empty schema.
                         </t>
                         <t>
-                            Implementation MAY choose to implement or optimize this keyword
+                            Implementations MAY choose to implement or optimize this keyword
                             in another way that produces the same effect, such as by directly
                             checking the names in "properties" and the patterns in
                             "patternProperties" against the instance property set.
                             Implementations that do not support annotation collection MUST do so.
                         </t>
                     </section>
 
+                    <section title="unevaluatedProperties" anchor="unevaluatedProperties">
+                        <t>
+                            The value of "unevaluatedProperties" MUST be a valid JSON Schema.
+                        </t>
+                        <t>
+                            The behavior of this keyword depends on the annotation results of
+                            adjacent keywords that apply to the instance location being validated.
+                            Specifically, the annotations from "properties", "patternProperties",
+                            and "additionalProperties", which can come from those keywords when
+                            they are adjacent to the "unvevaluatedProperties" keyword.  Those
+                            three annotations, as well as "unevaluatedProperties", can also
+                            result from any and all adjacent
+                            <xref target="in-place">in-place applicator</xref> keywords.
+                            This includes but is not limited to the in-place applicators
+                            defined in this document.
+                        </t>
+                        <t>
+                            Validation with "unevaluatedProperties" applies only to the child
+                            values of instance names that do not appear in the "properties",
+                            "patternProperties", "additionalProperties", or
```

Comment:
I can add that, yeah.  Better to be clear.

Reply:
@Relequestual actually no, this is correct, because it is talking about the names present in the "patternProperties" **annotation results**.  Which are actual _instance_ property names, not patterns.  They are already the names that matched, as the matching happened earlier, when `patternProperties` itself was evaluated.

Reply:
The annotation results of the relevant keywords (`properties`, `patternProperties`, `additionalProperties`) are defined in the description of those keywords. Please take a look and see if it that answers your question, as that as how most spec readers will be figuring it out.